### PR TITLE
Adds podmonitors for KCM, scheduler, authenticator pods

### PR DESCRIPTION
**Pull request by prateekgogia**
Issue #, if available:

Description of changes:
```
podMonitor/default/foo-apiserver/0 (1/1 up)
podMonitor/default/foo-authenticator/0 (1/1 up)
podMonitor/default/foo-controller-manager/0 (1/1 up)
podMonitor/default/foo-etcd/0 (3/3 up)
podMonitor/default/foo-scheduler/0 (1/1 up)
```
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: operator/pkg/controllers/master/kubecontrollermanager.go [hunk 1]
```diff
@@ -96,6 +93,10 @@ func kcmPodSpecFor(controlPlane *v1alpha1.ControlPlane) v1.PodSpec {
 					v1.ResourceCPU: resource.MustParse("1"),
 				},
 			},
+			Ports: []v1.ContainerPort{{
+				ContainerPort: 10252,
```
Comment (felix-zhe-huang) [inline operator/pkg/controllers/master/kubecontrollermanager.go:97]: I thought the insecure port has been deprecated in 1.19? what happen if we choose to use `KubernetesVersion=v1.21`?

Comment (felix-zhe-huang) [inline operator/pkg/controllers/master/kubecontrollermanager.go:97]: I thought the insecure port has been deprecated in 1.19? what happen if we choose to use `KubernetesVersion=v1.21`?

File: operator/pkg/controllers/master/kubescheduler.go [hunk 2]
```diff
@@ -85,6 +81,10 @@ func schedulerPodSpecFor(controlPlane *v1alpha1.ControlPlane) v1.PodSpec {
 					v1.ResourceCPU: resource.MustParse("1"),
 				},
 			},
+			Ports: []v1.ContainerPort{{
+				ContainerPort: 10251,
```
Comment (felix-zhe-huang) [inline operator/pkg/controllers/master/kubescheduler.go:85]: same problem. What happen if we choose to use KubernetesVersion=v1.21?

Comment (felix-zhe-huang) [inline operator/pkg/controllers/master/kubescheduler.go:85]: same problem. What happen if we choose to use KubernetesVersion=v1.21?

File: operator/pkg/controllers/master/podmonitors.go [hunk 3]
```diff
@@ -82,3 +86,35 @@ func etcdPodMonitorFor(controlPlane *v1alpha1.ControlPlane) monitoringv1.PodMoni
...
+
+func authenticatorPodMonitorFor(controlPlane *v1alpha1.ControlPlane) monitoringv1.PodMonitorSpec {
+	return monitoringv1.PodMonitorSpec{
+		JobLabel:          fmt.Sprintf("%s-authenticator", controlPlane.ClusterName()),
+		NamespaceSelector: monitoringv1.NamespaceSelector{MatchNames: []string{controlPlane.Namespace}},
+		Selector:          metav1.LabelSelector{MatchLabels: iamauthenticator.Labels},
+		PodMetricsEndpoints: []monitoringv1.PodMetricsEndpoint{{
+			Port: "metrics", Scheme: "https",
+			TLSConfig: &monitoringv1.PodMetricsEndpointTLSConfig{
+				SafeTLSConfig: monitoringv1.SafeTLSConfig{InsecureSkipVerify: true},
```
Comment (felix-zhe-huang) [inline operator/pkg/controllers/master/podmonitors.go:116]: Is it possible to use 
`SafeTLSConfig: monitoringv1.SafeTLSConfig{CA: IAMAUTHENTICATOR_CA}`?

Comment (felix-zhe-huang) [inline operator/pkg/controllers/master/podmonitors.go:116]: Is it possible to use 
`SafeTLSConfig: monitoringv1.SafeTLSConfig{CA: IAMAUTHENTICATOR_CA}`?

File: operator/pkg/controllers/master/kubecontrollermanager.go (hunk 1 above)
Comment (prateekgogia) [reply operator/pkg/controllers/master/kubecontrollermanager.go:97]: > I thought the insecure port has been deprecated in 1.19?
Thats for API server only, KCM and scheduler is not yet deprecated

File: operator/pkg/controllers/master/kubescheduler.go (hunk 2 above)
Comment (prateekgogia) [reply operator/pkg/controllers/master/kubescheduler.go:85]: In 1.21 KCM and scheduler is not yet deprecated

File: operator/pkg/controllers/master/podmonitors.go (hunk 3 above)
Comment (prateekgogia) [reply operator/pkg/controllers/master/podmonitors.go:116]: CA for authenticator is not saved anywhere but only in its pod, so we can't access it at the moment.

**Review by felix-zhe-huang — APPROVED** [review]
LGTM. Ship it!

**Pull request merged**